src/charm.py

```diff
@@ -120,11 +168,55 @@ def __init__(self, *args):
         observe(ipu_events.data_provided, self._handle_ingress_data_provided)
         observe(ipu_events.data_removed, self._handle_ingress_data_removed)
 
-        observe(self.traefik_route.on.ready, self._handle_traefik_route_ready)
+        ipr_events = self.traefik_route.on
```

Comment:
"ipr" (ingress per route?) isn't a very meaningful name considering what the charm/relation actually do. Could we change this to `tr_events` or `route_events`?